plugins/database/src/db_plugin.cpp

```diff
@@ -14920,6 +14922,165 @@ auto db_get_delay_rule_info_op(irods::plugin_context& _ctx, const char* _rule_id
+
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-constant-array-index)
+            cllBindVars[cllBindVarCount++] = get_json_string_value(after, "data_repl_num");
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-constant-array-index)
+            cllBindVars[cllBindVarCount++] = get_json_string_value(after, "data_version");
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-constant-array-index)
+            cllBindVars[cllBindVarCount++] = get_json_string_value(after, "data_type_name");
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-constant-array-index)
+            cllBindVars[cllBindVarCount++] = get_json_string_value(after, "data_size");
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-constant-array-index)
+            cllBindVars[cllBindVarCount++] = get_json_string_value(after, "data_path");
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-constant-array-index)
+            cllBindVars[cllBindVarCount++] = get_json_string_value(after, "data_owner_name");
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-constant-array-index)
+            cllBindVars[cllBindVarCount++] = get_json_string_value(after, "data_owner_zone");
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-constant-array-index)
+            cllBindVars[cllBindVarCount++] = get_json_string_value(after, "data_is_dirty");
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-constant-array-index)
+            cllBindVars[cllBindVarCount++] = get_json_string_value(after, "data_status");
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-constant-array-index)
+            cllBindVars[cllBindVarCount++] = get_json_string_value(after, "data_checksum");
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-constant-array-index)
+            cllBindVars[cllBindVarCount++] = get_json_string_value(after, "data_expiry_ts");
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-constant-array-index)
+            cllBindVars[cllBindVarCount++] = get_json_string_value(after, "data_map_id");
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-constant-array-index)
+            cllBindVars[cllBindVarCount++] = get_json_string_value(after, "data_mode");
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-constant-array-index)
+            cllBindVars[cllBindVarCount++] = get_json_string_value(after, "r_comment");
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-constant-array-index)
+            cllBindVars[cllBindVarCount++] = get_json_string_value(after, "create_ts");
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-constant-array-index)
+            cllBindVars[cllBindVarCount++] = get_json_string_value(after, "modify_ts");
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-constant-array-index)
+            cllBindVars[cllBindVarCount++] = get_json_string_value(after, "resc_id");
```

Comment:
Oh god. Consider wrapping these lines in `NOLINTBEGIN(...)` and `NOLINTEND(...)` so that it's not as loud.

ClangTidy won't understand it at this time, but it will once we decide to bump the compiler (in the future).

Reply:
Okay. I wanted to use those but wasn't sure if using the future-things was acceptable. That being said, in order to avoid my inevitable screwing something up by changing it, I think I'm going to leave it as-is.

Reply:
That's fine.

Reply:
I promise it'll go away one day ;)